providers/common.rb

```diff
@@ -0,0 +1,71 @@
+include_recipe 'git'
+
+action :install do
+
+  if new_resource.create_user
+    group new_resource.group do
+      action :create
+    end
+    user new_resource.owner do
+      action :create
+      gid new_resource.group
+    end
+  end
+
+  # If the new_resource.path is nil set the install path to
+  # `/opt/name_attribute`
+  if new_resource.path.nil?
+    new_resource.path = "/opt/#{ new_resource.name_attribute }"
+  end
+
+  directory new_resource.path do
+    action :create
+    owner new_resource.owner
+    group new_resource.group
+  end
+
+  # Create virtual environment
+  python_virtualenv new_resource.virtualenv_path do
```

Comment:
Does the `python_virtualenv` resource make sure that virtualenv is installed? Otherwise you might want to include the python recipes first.